doc/source/analyzing/analysis_modules/absorption_spectrum.rst

```diff
@@ -195,7 +195,7 @@ errors produced in saving to a ascii file will negatively impact fit quality.
 
 .. code-block:: python
 
-    f = h5py.File('spectrum.h5')
+    f = h5py.File('spectrum.h5', 'r')
```

Comment:
Since this API is prone to changes, which is the very motivation for your PR, I would recommend using the explicit form
```suggestion
    f = h5py.File('spectrum.h5', mode='r')
```

Reply:
That would be a quite larger change than that: a quick count that `h5py.File`  is used >150 times in yt (and everywhere with the mode as positional argument).

Reply:
Here's a script that can do it for you !

```python
import re
from pathlib import Path

exp = re.compile("h5py\.File\([^,]+,\s*")
yt_dir = Path(__file__).parent
for file in yt_dir.glob("**/*.py"):
    matches = None
    with open(file, mode="r") as fileobj:
        text = fileobj.read()
        matches = re.findall(exp, text)
        for match in matches:
            print(type(match))
            text = text.replace(match, match + " mode=")
    if matches:
        with open(file, mode="w") as fileobj:
            fileobj.write(text)
```

save it as `yt/magic.py` and run `python magic.py`, then you can validate that it did the correct thing with `git diff` (and `git stash` to revert immediately in case something went wrong). I tested it on my machine and it worked :)

Reply:
In case you are using email-notifications: sorry, my first attempt was actually broken, the script should work now !

Reply:
How about if just the changes you've made in this PR use `mode`, @olebole ? We could issue a separate PR to update the use in the rest of the codebase (we're not consistent with the usage of `mode` already, so I think it's fine to not update the other lines). That script looks great though @neutrinoceros!

Reply:
(side-note: the script is idempotent as it is, it doesn't require that all occurrences be changed at the same time !)